src/test/scheduler/scheduler_test.cpp

```diff
@@ -126,6 +126,31 @@ TEST_F(SchedulerTest, BasicTest) {
   CurrentScheduler::set(nullptr);
 }
 
+TEST_F(SchedulerTest, PendingTasks) {
+  Topology::use_fake_numa_topology(8, 4);
+  CurrentScheduler::set(std::make_shared<NodeQueueScheduler>());
+
+  std::atomic_uint counter{0};
+
+  auto task = std::make_shared<JobTask>([&]() {
+    // Wait for test to increment the counter, then increment it again
+    while (counter == 0) {
+    }
+    ++counter;
+  });
+  task->schedule();
+
+  ASSERT_TRUE(CurrentScheduler::has_pending_tasks());
```

Comment:
This test confuses pending tasks and running tasks. The change to the BenchmarkRunner should also check for both. Will rework this. Don't let this stop you from reviewing the rest.